packages/denylist-bot/src/utils.ts

```diff
@@ -0,0 +1,65 @@
+import Process from "process";
+import { Buffer } from "buffer";
+import type { Env } from "./types/env";
+import { connect, keyStores, KeyPair, providers } from "near-api-js";
+
+/* tslint:disable */
+globalThis.Buffer = Buffer;
+globalThis.process = Process;
+/* tslint:enable */
+
+export const AttachedGas = "300000000000000";
+
+export const setupNear = async ({
+  RPC_URL,
+  ACCOUNT_ID,
+  PRIVATE_KEY,
+  NEAR_NETWORK,
+}: Env) => {
+  const myKeyStore = new keyStores.InMemoryKeyStore();
+
+  const keyPair = KeyPair.fromString(PRIVATE_KEY);
+
+  await myKeyStore.setKey(NEAR_NETWORK, ACCOUNT_ID, keyPair);
+
+  const connectionConfig = {
+    nodeUrl: RPC_URL,
+    networkId: NEAR_NETWORK,
+  };
+
+  return connect({ ...connectionConfig, keyStore: myKeyStore });
+};
+
+export const viewFunction = async (
+  nodeUrl: string,
+  contractId: string,
+  methodName: string,
+  args: any = {}
+) => {
+  const provider = new providers.JsonRpcProvider({ url: nodeUrl });
+
+  const serializedArgs = Buffer.from(JSON.stringify(args)).toString("base64");
+
+  const res = (await provider.query({
+      request_type: "call_function",
+      account_id: contractId,
+      method_name: methodName,
+      args_base64: serializedArgs,
+      finality: "optimistic",
+  })) as any;
+
+  return JSON.parse(Buffer.from(res.result).toString());
+};
+
+export const paginationSize = 100;
+
+export const getHapioneEntriesQuery = `
+  query GetHapioneEntries($lastViewed: BigInt!) {
+    hapioneEntries(where: {counter_gt: $lastViewed}, first: ${paginationSize}, orderBy: counter) {
```

Comment:
Shouldn't counter be greater than OR EQUAL to `$lastViewed`? Consider the initial case where `$lastViewed` = 0 because no query has yet been made. We want it to fetch the entry of counter 0.

Likewise, if we query 50 entries (0 to 49) we are going to ser `$lastViewed` to 50, so that it starts the query at 50 next time.

If we only use greater than we are not going to achieve that effect.